ldap/servers/plugins/replication/repl5_agmt.c

```diff
@@ -482,8 +482,20 @@ agmt_new_from_entry(Slapi_Entry *e)
 
     /* DBDB: review this code */
     if (slapi_entry_attr_hasvalue(e, "objectclass", "nsDSWindowsReplicationAgreement")) {
-        ra->agreement_type = REPLICA_TYPE_WINDOWS;
-        windows_init_agreement_from_entry(ra, e);
+        if (replica_get_type(replica) == REPLICA_TYPE_PRIMARY) {
```

Comment:
type PRIMARY (1) or UPDATABLE (3) should have a CL and we can have a RA.
Shouldn't we fail only for READONLY replica, else create the RA ?

Reply:
But it's then valid to have a topology which is:

```
PRIMARY -> HUB <- WINSYNC
```

Which means that HUB no longer is consistent, nor is the topology. All changes from winsync must go to a full supplier.

Reply:
replicaType (lib389) comes from 'nsDS5ReplicaType'. It looks lib389 only supports REPLICA_TYPE_UPDATABLE and REPLICA_TYPE_READONLY. I do not know where the value REPLICA_TYPE_PRIMARY is coming from (during config).

Anyway we want to create/init RA if there is a CL. It exists CL for REPLICA_TYPE_UPDATABLE and possibly also for REPLICA_TYPE_PRIMARY. 
So I wonder if the test should rather be 
<pre>
if (replica_get_type(replica) == REPLICA_TYPE_PRIMARY || replica_get_type(replica) == REPLICA_TYPE_UPDATABLE) {
...
}
</pre>

Reply:
I literally tested this. Primary is when it is a supplier. When we set this as a hub it's NOT a primary, and IS uptable. It works. 

As mentioned, we *can not* match on updateable here because that's an invalid replication topology.

Reply:
Sorry @Firstyear for my slow understanding. I missed that HUB case is an invalid replication topology. So this change is good to me.

Side note, IIUC this part of code is to handle outgoing RA to windows. It is not clear to me why hub can not act as a supplier.

Side note, during upgrade (_cl5_upgrade_replica), I do not find any specificity supplier vs hub. It looks to me that in both cases the changelog file and config is move to the main DB.

Reply:
It handles incoming replication too @tbordaz .... it does both directions. 

The upgrade issue is for consumer vs supplier, not hub.